py-polars/polars/dataframe/plotting.py

```diff
@@ -0,0 +1,163 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Any
+
+if TYPE_CHECKING:
+    import altair as alt
+
+    from polars import DataFrame
+
+
+class Plot:
+    """DataFrame.plot namespace."""
+
+    chart: alt.Chart
+
+    def __init__(self, df: DataFrame) -> None:
+        import altair as alt
+
+        self.chart = alt.Chart(df)
+
+    def line(
+        self,
+        x: str | Any | None = None,
+        y: str | Any | None = None,
+        color: str | Any | None = None,
+        order: str | Any | None = None,
+        tooltip: str | Any | None = None,
+        *args: Any,
+        **kwargs: Any,
+    ) -> alt.Chart:
```

Comment:
Hi @dangotbanned - may I ask for your input here please?

1. which do you think are the most common types of plots which are worth explicitly making functions for? Functionality would be unaffected, they would just work better with tab completion
2. how would you suggest typing the various arguments? Does Altair have public type hints?

Thanks 🙏

Reply:
Thanks for the ping, happy to help where I can @MarcoGorelli 

Couple of resources up top that I think could be useful:
- [Examples Gallery](https://altair-viz.github.io/gallery/index.html#example-gallery)
- [An archived version of this idea](https://github.com/altair-viz/pdvega/blob/e3f1fc9730f8cd9ad70e7ba0f0a557f41279839a/pdvega/_core.py#L374-L1031)
- [User Guide - Marks](https://altair-viz.github.io/user_guide/marks/index.html#marks)
- [User Guide - long vs wide form](https://altair-viz.github.io/user_guide/data.html#long-form-vs-wide-form-data)

Will respond each question in another comment 👍

Reply:
> 1. which do you think are the most common types of plots which are worth explicitly making functions for? Functionality would be unaffected, they would just work better with tab completion

Can't speak for everyone, but for a reduced selection:

- [Bar](https://altair-viz.github.io/user_guide/marks/bar.html#bar)
- Line ([Line](https://altair-viz.github.io/user_guide/marks/line.html), [Trail](https://altair-viz.github.io/user_guide/marks/trail.html#line-chart-with-varying-size))
- Scatter ([Circle](https://altair-viz.github.io/user_guide/marks/circle.html#scatter-plot-with-circle), [Point](https://altair-viz.github.io/user_guide/marks/point.html#scatter-plot), [Square](https://altair-viz.github.io/user_guide/marks/square.html), [Image](https://altair-viz.github.io/user_guide/marks/image.html), [Text](https://altair-viz.github.io/user_guide/marks/text.html))
- Area ([Area](https://altair-viz.github.io/user_guide/marks/area.html#area), [Rect](https://altair-viz.github.io/user_guide/marks/rect.html#examples))
- [Tick](https://altair-viz.github.io/user_guide/marks/tick.html#examples)
- [Boxplot](https://altair-viz.github.io/user_guide/marks/boxplot.html#box-plot)
- [Geoshape](https://altair-viz.github.io/user_guide/marks/geoshape.html#geoshape)
	- Field-dependent, but super important for those who need it

Looking at [hvPlot](https://github.com/holoviz/hvplot/blob/051378e62ba50a67b2ea409aee500604d54c9d5b/hvplot/plotting/core.py#L1985-L2013), there are a few methods/chart types I'd need to do some digging to work out the equivalent in `altair` (if there is one).

However, my suggestion would be using the names defined there, both for compatibility when switching backends and to reduce the number of methods.

# Examples
Haven't covered everything here, but it's a start:

`hvPlotTabular` -> `altair.Chart`
- `(bar|barh)` -> `mark_bar`
- `box` -> `mark_boxplot`
- `scatter` -> `mark_(circle|point|square|image|text)`
  - `labels` -> `mark_text` 
  - `points` -> `mark_point`
- `line` ->` mark_(line|trail)`
- `(polygons|paths)` -> `mark_geoshape`
- `(area|heatmap)` -> `mark_area`

Reply:
# 2
> 2. how would you suggest typing the various arguments? Does Altair have public type hints?
>
I might update this later after thinking on it some more.

Yeah they've been there since [`5.2.0`](https://github.com/vega/altair/releases/tag/v5.2.0) but will be improved for `altair>=5.4.0` with https://github.com/vega/altair/blob/main/altair/vegalite/v5/schema/_typing.py

For `altair` the model is quite different to `matplotlib`-based style functions, but [`.encode()`](https://github.com/vega/altair/blob/d2325e3dd2a2af9156fcb1a1e3a3ff655a31128e/altair/vegalite/v5/schema/channels.py#L30901) would be where to start.

Something like:

```py
# Annotation from `.encode()`
# y: Optional[str | Y | Map | YDatum | YValue] = Undefined

# Don't name it this pls
TypeForY = str | Mapping[str, Any] | Any
```

I wouldn't worry about any `altair`-specific types here. 
Spelling them out won't have an impact on attribute access of the result

Reply:
# 3
> 3. Any Altair maintainers you'd suggest looping into the discussion?

For typing @binste but really anyone from https://github.com/vega/altair/pull/3452 I think would be interested (time-permitting)

@mattijn, @joelostblom, @jonmmease